enterprise/dbcrypt/dbcrypt.go

```diff
@@ -0,0 +1,296 @@
+// Package dbcrypt provides a database.Store wrapper that encrypts/decrypts
+// values stored at rest in the database.
+//
+// Encryption is done using a Cipher. The Cipher is stored in an atomic pointer
+// so that it can be rotated as required.
+//
+// The Cipher is currently used to encrypt/decrypt the following fields:
+// - database.UserLink.OAuthAccessToken
+// - database.UserLink.OAuthRefreshToken
+// - database.GitAuthLink.OAuthAccessToken
+// - database.GitAuthLink.OAuthRefreshToken
+// - database.DBCryptSentinelValue
+//
+// Encrypted fields are stored in the following format:
+// "dbcrypt-${b64encode(<first 7 digits of cipher's SHA256 digest>-<encrypted value>)}"
```

Comment:
Rolling our own fixed-width format here might cause headaches in the future.  What do you think about, say, making this a protobuf?

Reply:
We should keep the magic prefix, but I'm open to storing the data as a protobuf. We will probably still need to base64-encode it though.

Reply:
yeah.  Even if we picked a text-based protocol, like JSON, the encrypted payload is binary, so we'd be b64 encoding the bulk anyway.